source/common/res/features/budget-to-goal/main.js

```diff
@@ -0,0 +1,163 @@
+(function poll() {
+  if (typeof ynabToolKit !== 'undefined' && ynabToolKit.pageReady === true) {
+    ynabToolKit.budgetBalanceToGoal = (function () {
+      return {
+        budgetView: ynab.YNABSharedLib.getBudgetViewModel_AllBudgetMonthsViewModel()._result,
+
+        invoke() {
+          let categories = ynabToolKit.budgetBalanceToGoal.getCategories();
+          let categoryName = ynabToolKit.budgetBalanceToGoal.getInspectorName();
+          let masterCategoryViewId = $('ul.is-checked').prevAll('ul.is-master-category').attr('id');
+          let masterCategory = ynabToolKit.shared.getEmberView(masterCategoryViewId).get('data');
+          let masterCategoryId = masterCategory.get('categoryId');
+
+          categories.forEach(function (f) {
+            if (f.name === categoryName && f.masterCategoryId === masterCategoryId) {
+              ynabToolKit.budgetBalanceToGoal.updateInspectorButton(f);
+
+              return false;
+            }
+          });
+        },
+
+        observe(changedNodes) {
+          console.log(changedNodes);
+          if (changedNodes.has('budget-inspector-goals')) {
+            // In theory, if there's a 'budget-inspector-goals' classed node, then the user is or has changed the goal.
+            // However, there seems to be some disconnect between this event and when YNAB updates the values.
```

Comment:
Watching the Goal Target button might work for Monthly or Target by Date, but not so much for Target without Date (no Goal Target). Target with and without date are the primary targets [sic] for this. I will probably hide the button for Monthly, since it is pointless.

Going along the same lines, perhaps I can directly monitor the "To Go" value, though it wouldn't allow for negative budgeting. This might get me away from the "._result" as well.